### PR TITLE
Give editable fields in blocks better aria-labels.

### LINKED ISSUE #1659: Give editable fields better labels
Splitting this out from #1477 

Screen readers and other assistive technologies (for example speech recognition software) use the `aria-label` attribute to announce the name (the "accessible name") of UI controls to users.

`contenteditable` regions in the editor use an `aria-label` attribute and a visible placeholder text with the same value. For example, a quote block uses a `Write quote…` text for both.

One important thing to consider is that `contenteditable` regions tend to be reported by screen readers as sort of form fields. The behavior differs across screen readers but they understand it's an editable region. Thus, an `aria-label` gets announced as it was the label of a form field:

screen shot 2017-07-03 at 11 54 58

NVDA instead tends to announce as `editable` the actual HTML element, for example it announces:
`paragraph editable` for the text block
`list editable` for the list block
`text frame editable` for the button block (because it's a span)

Regardless, it does announce the aria-label. So what happens, in a few words?
There's the risk the visible placeholder text is reported as content of the editable text, so hearing, for example: `Write label ... Write label` should be avoided. One option could be hiding the visible placeholder from assistive technologies with `aria-hidden`.

However, I'm not sure we should use `Write label…`, `Write…`, etc., as aria-label in the first place. They behave like labels of a form field so they are announced even after users enter some content, for example:

screen shot 2017-07-03 at 11 55 35
screen shot 2017-07-03 at 11 56 01

This could be confusing, as the aria-label text in this scenario is not so appropriate. Wondering how to improve the overall way editable regions are exposed to assistive technologies. Any feedback welcome!

### PR DESCRIPTION
**Pull request by ZebulanStanphill**
## Description
Fixes #1659.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by gziolo** [comment]
It looks like this PR needs testing and should be good to go.

File: packages/block-library/src/shortcode/edit.js
Comment (gziolo) [inline packages/block-library/src/shortcode/edit.js:26]: "text" might be not the best fit for shortcodes. How about 'Shortcode content'?

File: packages/block-library/src/verse/edit.js
Comment (gziolo) [inline packages/block-library/src/verse/edit.js:50]: In addition to that, should we align `placeholder`?
```suggestion
				placeholder={ __( 'Write verse…' ) }
```

File: packages/block-library/src/post-author/edit.js
Comment (gziolo) [inline packages/block-library/src/post-author/edit.js:148]:
```suggestion
							placeholder={ __( 'Write byline…' ) }
```

File: packages/block-library/src/site-tagline/edit.js
Comment (gziolo) [inline packages/block-library/src/site-tagline/edit.js:45]:
```suggestion
				placeholder={ __( 'Write site tagline …' ) }
```

File: packages/block-library/src/site-title/edit/index.js
Comment (gziolo) [inline packages/block-library/src/site-title/edit/index.js:54]:
```suggestion
				placeholder={ __( 'Write site title…' ) }
```

**Review by gziolo — APPROVED** [review]
Proposed changes look good 👍 

Ideally, it should be tested by someone using assistive technologies before merging.

I also included some suggestions for `placeholder` props to align with the most popular usage. Totally optional, I leave it up to you to decide whether they make sense to apply 😄

File: packages/block-library/src/shortcode/edit.js
Comment (ZebulanStanphill) [reply packages/block-library/src/shortcode/edit.js:26]: "Content" sounds like it might refer to stuff between shortcode tags, but not including them, so I'm not sure if it's the right term. What do you think, @afercia?

**Comment by gziolo** [comment]
Thank you @ZebulanStanphill for addressing my feedback, feel free to merge based on my review if there is no more feedback added 😄

**Review by aristath — APPROVED** [review]
LGTM :+1:

**Pull request merged**
